### DISCUSSION TITLE
ImageGlass will support the apng format?

### DISCUSSION THREAD

**Discussion by Dhyfer1** [d2phap/ImageGlass#966] [Q&A]
Hi Dev.

I recently came across ImageGlass and it is now my default image viewer. It's lightweight, fast, beautiful and feature-rich 😃
Well, I am a graphic designer and I work with many types of image files, including animated png files or also known as apng files. ImageGlass has opened and correctly displayed every type of image I need, except my apng files (if I open an apng file with ImageGlass, it only shows it as any png file), which I don't understand, because if ImageGlass can open animated image files like .gif files, why can't it open .apng files?

Here [https://imageglass.org/docs/supported-formats](url) it says: 
_ImageGlass supports the most common image formats thanks to Magick.NET library. User can easily customize the extension list to add more file types and open in ImageGlass as long as Magick.NET supports it._

And here [https://github.com/d2phap/ImageGlass/wiki/Add-new-file-extensions](url) it says:
_ImageGlass relies on Magick.Net for image support. Sometimes, Magick.NET supports file formats which are not listed as supported by ImageGlass._

From the above, I understand that ImageGlass relies on the Magick.NET library to read the image formats supported by that library, and according to the table with the supported image formats that you can see here: [https://imagemagick.org/script/formats.php#supported](url) I find that the apng format is supported by the Magick.Net library, but within the ImageGlass settings, in the File Type Associations section, apng does not appear as a supported format, however I add it to the list because as I said before, it is supported by the library, but after customizing the list with the new file type and after opening an apng file, simply the apng file appears as any png file.

So my question is... does the developer plan to add .apng file support for ImageGlass? I hope so.
That's all, thanks in advance.

**Comment by d2phap** [comment]
Hi @Dhyfer1 

Thanks for the question.
I was looking for a solution to support animation for webp (and other animating format) in the past, probably by converting webp image frames into Bitmap object. Magick.Net provides methods to read all of frames of the image, as well as the timing meta data, but I didn't find an optimal way to convert to Bitmap object with less memory usage.
I will continue on my investigation.

Phap

**Reply by fire-eggs** [reply]
The top answer to this [StackOverflow question](https://stackoverflow.com/questions/31675557/creating-an-animated-bitmap-image-programmatically) shows a method to create an `Image` object from a set of frames and timing data.

**Reply by Dhyfer1** [reply]
> I was looking for a solution to support animation for webp (and other animating format) in the past, probably by converting webp image frames into Bitmap object. Magick.Net provides methods to read all of frames of the image, as well as the timing meta data, but I didn't find an optimal way to convert to Bitmap object with less memory usage.
> I will continue on my investigation.

Hi again.

Ok, I hope you find something about how to implement apng file support for ImageGlass, I know you will find something later for that. For my part I leave you these links, I hope that some of these can help you about apng support:

- [https://github.com/jz5/APNG.NET](url)
- https://github.com/dlemstra/Magick.NET
- https://github.com/xupefei/APNG.NET
- https://github.com/rickytan/libpng-apng
- https://github.com/murrple-1/APNGManagement
- https://github.com/smoked-herring/sail

By the way, I have a question. Magick.NET is the library used by the ImageMagick program, and on the ImageMagick home page it says:
_"Use ImageMagick® to create, edit, compose, or convert bitmap images. It can read and write images in a variety of formats (over 200) including PNG, JPEG, GIF, HEIC, TIFF, DPX, EXR, WebP, Postscript, PDF, and SVG."_

So I've been wondering, if ImageGlass and ImageMagick use the Magick.NET library, is it possible that ImageGlass can acquire the function of converting bitmap images like ImageMagick does? or is that function (converting bitmap images) independent of the library? Because I would like to see ImageGlass have the ability to convert my images from one format to another, it would be very useful. 

That's it, I hope for your answer.

**Reply by Dhyfer1** [reply]
> The top answer to this [StackOverflow question](https://stackoverflow.com/questions/31675557/creating-an-animated-bitmap-image-programmatically) shows a method to create an `Image` object from a set of frames and timing data.

Thanks, but I'm not a developer or programmer, so I'm using [QuickLook](https://github.com/QL-Win/QuickLook) for file viewing. With that I can visualize apng files until ImageGlass can do it.

**Reply by d2phap** [reply]
@Dhyfer1 FYI, You can play animated PNG images (.APNG, .PNG) in ImageGlass 10 with [APNG codec plugin](https://github.com/d2phap/apng-imageglass-plugin)

**Reply by Dhyfer1** [reply]
Wow. Thank you so much for the plugin, it works great and now I can finally see my .apng files correctly. 👍
